### PR TITLE
Afpacket improvements/v13

### PR DESCRIPTION
**Pull request by victorjulien**
Continuation of #6544 work.

https://redmine.openinfosecfoundation.org/issues/4785
https://redmine.openinfosecfoundation.org/issues/4796
https://redmine.openinfosecfoundation.org/issues/4800

### CODE REVIEW

File: src/source-af-packet.c [hunk 1]
```diff
@@ -169,10 +169,22 @@ TmEcode NoAFPSupportExit(ThreadVars *tv, const void *initdata, void **data)
...
 #ifndef TP_STATUS_USER_BUSY
-/* for new use latest bit available in tp_status */
-#define TP_STATUS_USER_BUSY     BIT_U32(31)
+/* HACK special setting in the tp_status field for frames we are
+ * still working on. This can happen in autofp mode where the
+ * capture thread goes around the ring and finds a frame that still
+ * hasn't been released by a worker thread.
+ *
+ * We use bits 29, 30, 31. 29 and 31 are software and hardware
+ * timestamps. 30 should not be used. Combined they should never
```
Comment (victorjulien) [inline src/source-af-packet.c:183]: kernel says it should never set this
```
#define TP_STATUS_TS_SYS_HARDWARE        (1 << 30) /* deprecated, never set */
```

**Comment by victorjulien** [comment]
Flow counter commit doesn't belong here.

First afpacket commit should start with `af-packet:`

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.wrk.flows_evicted | 5304431 | 4693221 | 113.02% |
| .flow.mgr.rows_maxlen | 115 | 539 | 21.34% | 

Pipeline 4676

**Comment by victorjulien** [comment]
Replaced by #6553

**Pull request closed without merging**